### PR TITLE
[CF-551] Extend functional tests to verify tenant mapping functionality

### PR DESCRIPTION
**Pull request by jaspreetrawel**
Added the below scenario:

-     Create tenant "tenant2" in source cloud
-     Map "tenant2" to "tenant-2" in configs/resource_map.yaml mapping file
-     Create image, network, VM with attached volume in "tenant2" tenant
-     Run migration
-     Verify all created objects successfully migrated to destination tenant "tenant-2"

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by jaspreetrawel** [comment]
Tested in local environment, meanwhile testing with vagrant box grizzly to icehouse job with DEBUG ON and then functional test grizzly to juno job with DEBUG ON.

File: cloudferry_devlab/cloudferry_devlab/tests/functional_test.py [hunk 1]
```diff
@@ -72,6 +72,12 @@ def setUp(self):
         self.filtering_utils = utils.FilteringUtils(
             os.path.join(self.cloudferry_dir, filter_path))
 
+    def get_mapped_tenant(self, tenant_name):
+        tnt_name = config.mapped_tenant_dict[tenant_name] \
+            if tenant_name in config.mapped_tenant_dict.keys() \
+            else tenant_name
```
Comment (roman-verchikov) [inline cloudferry_devlab/cloudferry_devlab/tests/functional_test.py]: Maybe

```python
return config.mapped_tenant_dict.get(tenant_name, tenant_name)
```

File: cloudferry_devlab/cloudferry_devlab/tests/base.py [hunk 2]
```diff
@@ -290,6 +290,14 @@ def get_sec_group_id_by_tenant_id(self, tenant_id):
         return [i['id'] for i in sec_group_list['security_groups']
                 if i['tenant_id'] == tenant_id]
 
+    def check_mapped_tenant(self, tenant_name, skip_cloud_prefix=False):
+        tnt_name = tenant_name
+        if skip_cloud_prefix or self.cloud_prefix == 'dst':
+            tnt_name = self.config.mapped_tenant_dict[tenant_name] \
+                if tenant_name in self.config.mapped_tenant_dict.keys() \
+                else tenant_name
```
Comment (roman-verchikov) [inline cloudferry_devlab/cloudferry_devlab/tests/base.py]: you have the same code in `FunctionalTest.get_mapped_tenant()`, please reuse

File: cloudferry_devlab/cloudferry_devlab/tests/config.py [hunk 3]
```diff
@@ -237,7 +244,8 @@
          {'name': 'tn1fl1', 'disk': '1', 'ram': '64', 'vcpus': '1'}
      ]
      },
-    {'name': 'tenant2', 'description': 'Bljakslhf ajsdfh', 'enabled': True,
+    {'name': 'tenant2', 'description': 'Bljakslhf ajsdfh',
```
Comment (roman-verchikov) [inline cloudferry_devlab/cloudferry_devlab/tests/config.py]: Unnecessary change, please revert

File: cloudferry_devlab/cloudferry_devlab/tests/config.py [hunk 4]
```diff
@@ -70,6 +70,13 @@
 all_tenants_filter_filename = 'all_tenants_filter.yaml'
 """All tenants filter file."""
 
+rsc_map_filename = 'configs/resource_map.yaml'
+"""Resource Map file."""
+
+mapped_tenant_dict = {'tenant2': 'tenant-2'}
```
Comment (roman-verchikov) [inline cloudferry_devlab/cloudferry_devlab/tests/config.py]: Let's have some more descriptive tenant name instead of `tenant-2`, for example, `tenant2_mapped_in_resource_map_yaml`.

**Comment by roman-verchikov** [comment]
-1, please resolve inlines

File: cloudferry_devlab/cloudferry_devlab/tests/config.py (hunk 3 above)
Comment (jaspreetrawel) [inline cloudferry_devlab/cloudferry_devlab/tests/config.py]: Oops Did by mistake!! Reverted!!

File: cloudferry_devlab/cloudferry_devlab/tests/config.py (hunk 4 above)
Comment (jaspreetrawel) [inline cloudferry_devlab/cloudferry_devlab/tests/config.py]: Done!!

File: cloudferry_devlab/cloudferry_devlab/tests/functional_test.py (hunk 1 above)
Comment (jaspreetrawel) [inline cloudferry_devlab/cloudferry_devlab/tests/functional_test.py]: Right :) Can it be done!! Changed!!

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3939/

File: cloudferry_devlab/cloudferry_devlab/tests/base.py (hunk 2 above)
Comment (jaspreetrawel) [inline cloudferry_devlab/cloudferry_devlab/tests/base.py]: Moved to utils and used it at all places.

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by jaspreetrawel** [comment]
@roman-verchikov all review comments incorporated.

**Comment by roman-verchikov** [comment]
Code review: +1.

We need to make sure this code doesn't break anything, so we'd have to resolve the problems with CI first.

Do not merge until icehouse to juno and grizzly to juno pass

**Comment by SVilgelm** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by jaspreetrawel** [comment]
@roman-verchikov @SVilgelm the [grizzly to juno migration failure](http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-juno/345/) is not due to the changes as part of this pull request. Those failures are due to override_rules.yaml missing the nfs2 to nfs_other override rule for volume types. This rule now exists in latest override_rules.yaml in SAIC gerrit project.

**Comment by plomakin** [comment]
+1

**Comment by roman-verchikov** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by roman-verchikov** [comment]
can be merged
